dds/DCPS/RTPS/ParameterListConverter.cpp

```diff
@@ -1284,7 +1290,7 @@ bool to_param_list(const DCPS::DiscoveredReaderData& reader_data,
     add_param(param_list, param);
   }
 
-  add_DataRepresentationQos(param_list, reader_data.ddsSubscriptionData.representation.value, true);
+  add_DataRepresentationQos(param_list, reader_data.ddsSubscriptionData.representation.value, true, true);
```

Comment:
How do we know the last parameter can always be `true`?
Another way to see the same issue - can the last parameter be removed?  It seems to only matter when the sequence is empty.  This function is called when transforming info about one of our local DataReaders to ParameterList format.  The caller could make sure that the sequence already accounts for the defaults -- before calling this we can know what we want to appear in the ParameterList in discovery, which may be an empty list (if we don't want anything to be encoded in the ParameterList) or not.